src/client/app/reducers/group.reducer.jsx

```diff
@@ -0,0 +1,32 @@
+let group = {
+  name: 'Loopy Leopards',
+  list: [
+    {
+      name: 'Giancarlo',
+      phone: '123-123-1234'
+    },
+    { 
+      name: 'Joe',
+      phone: '987 merge sort'
+    }
+  ]
+};
+
+export default function group(state = group, action) {
+  switch (action.type) {
+    case 'ADD_TO_GROUP':
+      state.list.push(action.payload);
```

Comment:
This is an interesting bit of syntax.  While returning an invocation of `Object.assign` is in fact going to make a new object, and thus satisfy Redux's need for returning a new object, `state.list.push` is still a mutative pattern-- it is going to mutate an old state object. This can be problematic for a few reasons: first, `Object.assign` uses a shallow copying algorithm.  This means that the object returned by `Object.assign` is in fact referencing the same array you've mutated here.  An easy fix for this is swapping out the `.push` method for the spread operator `[...state.list, action.payload]`, and moving this down into the `Object.assign` invocation, rather than modifying `state.list`.